dataset/operations.cpp

```diff
@@ -30,52 +30,102 @@ auto union_(const DatasetConstView &a, const DatasetConstView &b) {
   return out;
 }
 
-Dataset merge(const DatasetConstView &a, const DatasetConstView &b) {
+Dataset merge(const Dataset &a, const Dataset &b) {
   return Dataset(union_(a, b), union_(a.coords(), b.coords()));
 }
 
-/// Return a deep copy of a DataArray or of a DataArrayView.
-DataArray copy(const DataArrayConstView &array, const AttrPolicy attrPolicy) {
-  return DataArray(array, attrPolicy);
+Coords copy(const Coords &coords) { return {coords.sizes(), copy_map(coords)}; }
+Masks copy(const Masks &masks) { return {masks.sizes(), copy_map(masks)}; }
+
+/// Return a copy of a DataArray.
+DataArray copy(const DataArray &array, const AttrPolicy attrPolicy) {
+  // TODO is this correct? copy data but not meta data?
```

Comment:
Is this something the AttrPolicy could be extended to define? In any case sharing seems like the most sensible thing to me for now.

Reply:
Hmm, good suggestion. What would remain is to figure out what the default policy should be?

Reply:
I think generally data arrays should be thought of as "variable with meta data dicts", so if variable deep-copies by default, so should the data of a data array? Does that reasoning make sense?